model/src/main/kotlin/utils/CompositePackageConfigurationProvider.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.utils
+
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.config.PackageConfiguration
+
+/**
+ * A [PackageConfigurationProvider] that combines the provided [providers] into a single provider. The order of the
+ * [providers] determines the order in which they are accessed when requesting a [PackageConfiguration] in
+ * [getPackageConfiguration].
+ */
+class CompositePackageConfigurationProvider(
```

Comment:
I wonder whether we should somehow delimitate this more from `CompositePackageCurationProvider` to emphasize that this implementation does not accumulate / merge multiple `PackageConfiguration`s (which would be possible as long as they are not contradicting). This *could* simply be done by choosing another class name that does not include "composite". Because in the end, this is more like a `FallbackPackageCurationProvider` in that is uses the first match only.

On the other hand, package curations (in contrast to package configurations) were from the start designed to be cumulative, so maybe being more clear here is not needed after all. What do you think?

Reply:
Actually, in a later PR I want to change the interface to return all package configurations for a package, instead of only one. Now that we have multiple sources (global config and .ort.yml now, ort-config to be added) it makes a lot of sense to me, that you want to combine them. E.g. to amend the configurations from the ort-config repository with internal ones that can not be published.

Reply:
Ok. So maybe put in a `TODO` for now for clarity? Or is that upcoming PR so close that it's not worth it?

Reply:
It will come very soon. I first wanted to include the change in the PR that introduces the ort-config provider, but the latter requires some design discussion first so I will make them separately.

Reply:
The change was smaller than expected so I decided to add the commit to this PR as the reporter-web-app build was failing anyway for an unrelated reason.